### PR TITLE
Add POA Support

### PR DESCRIPTION
**Pull request by chinmay-browserstack**
CLI PR: https://github.com/percy/cli/pull/1250

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,3 +41,61 @@ module.exports = async function percySnapshot(driver, name, options) {
...
+    });
+  } catch (error) {
+    // Handle errors
+    log.error(`Could not take Screenshot "${name}"`);
+    log.error(error.stack);
+  }
+};
+
+const percy = module.exports = percySnapshot;
+percy.percyScreenshot = percyScreenshot;
```
Comment (samarsault) [inline index.js:101]: Can we do something like this
```javascript
module.exports.percyScreenshot = percyScreenshot
module.exports = percySnapshot
```

The above looks quite confusing

File: index.js [hunk 2]
```diff
@@ -39,3 +41,61 @@ module.exports = async function percySnapshot(driver, name, options) {
...
+  if (!name) throw new Error('The `name` argument is required.');
+  if (!(await utils.isPercyEnabled())) return;
+  let log = utils.logger('selenium-webdriver');
+
+  try {
+    let sessionId, capabilities, commandExecutorUrl;
+    if (driver.constructor.name === 'Browser') { // Logic for wdio
+      sessionId = driver.sessionId;
+      capabilities = driver.capabilities;
+      commandExecutorUrl = driver.options.protocol + '://' + driver.options.hostname + driver.options.path;
```
Comment (samarsault) [inline index.js:67]:
```
`${driver.options.protocol}://${driver.options.hostname}{driver.options.path}`
```

File: test/index.test.mjs [hunk 3]
```diff
@@ -62,3 +63,77 @@ describe('percySnapshot', () => {
     ]));
   });
 });
+
+describe('percyScreenshot', () => {
```
Comment (samarsault) [inline test/index.test.mjs:67]: Tests are failing and coverage is not 100%

File: index.js [hunk 4]
```diff
@@ -39,3 +41,61 @@ module.exports = async function percySnapshot(driver, name, options) {
     log.error(error);
   }
 };
+
+async function percyScreenshot(driver, name, options, postScreenshot = utils.postScreenshot) {
```
[the remark this answers is not in the document]
Comment (chinmay-browserstack) [reply index.js:45]: It is present in new CLI branch. updated description

File: test/index.test.mjs (hunk 3 above)
Comment (chinmay-browserstack) [reply test/index.test.mjs:67]: Screenshot 2023-05-29 at 6 07 25 PM
It is due to CLI changes not merged. getting 100% in local

File: index.js [hunk 5]
```diff
@@ -39,3 +41,62 @@ module.exports = async function percySnapshot(driver, name, options) {
...
+      const session = await driver.getSession();
+      sessionId = session.getId();
+      capabilities = Object.fromEntries(session.getCapabilities().map_);
+
+      // To intercept request from driver. used to get remote server url
+      const interceptor = new RequestInterceptor(withDefaultInterceptors.default);
+      interceptor.use((req) => {
+        const url = req.url.href;
+        commandExecutorUrl = url.split('/session')[0];
+      });
```
Comment (samarsault) [inline index.js:82]: Is this a synchronous operation?

Comment (chinmay-browserstack) [reply index.js:82]: yes

**Review by itsjwala — APPROVED** [review]

**Pull request merged**